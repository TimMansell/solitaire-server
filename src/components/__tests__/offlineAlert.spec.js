import { shallowMount } from '@vue/test-utils';
import OfflineAlert from '@/components/OfflineAlert.vue';

describe('OfflineAlert.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(OfflineAlert, {
      computed: {
        isOnline: () => false,
      },
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should show alert', () => {
    const wrapper = shallowMount(OfflineAlert, {
      computed: {
        isOnline: () => false,
      },
    });

    expect(wrapper.findAll('[data-test="offline-alert"]').exists()).toBe(true);
  });

  it('should not show alert', () => {
    const wrapper = shallowMount(OfflineAlert, {
      computed: {
        isOnline: () => true,
      },
    });

    expect(wrapper.findAll('[data-test="offline-alert"]').exists()).toBe(false);
  });
});
