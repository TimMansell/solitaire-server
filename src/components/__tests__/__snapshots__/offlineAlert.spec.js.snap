// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfflineAlert.vue matches snapshot 1`] = `<toast-stub msgs="Connection to the server has been lost." btntext="How can I fix this?" btnclick="[Function]" position="secondary" duration="0" data-test="offline-alert"></toast-stub>`;
